src/Gui/ElideCheckBox.cpp

```diff
@@ -29,44 +29,47 @@
 
 namespace Gui {
 
-const int CheckboxSpacing = 18;
-
 ElideCheckBox::ElideCheckBox(QWidget *parent)
-    : QCheckBox(parent) {
+    : QCheckBox(parent)
+    , m_IndicatorWidth(style()->pixelMetric(QStyle::PM_IndicatorWidth, nullptr, this))
+    , m_CheckBoxLabelSpacing(style()->pixelMetric(QStyle::PM_CheckBoxLabelSpacing, nullptr, this))
```

Comment:
Why is this calculated in the constructor and not in paint method? I think that methods from style are mostly intended to be used from the paint events, size hints, etc.

Reply:
to be saved and reused, it's used on both sizeHints, on an early version i used it for the paint event too, i can make the individual calls in both sizeHint functions if you prefer that

Reply:
It potentially could become problematic if we use such checkbox and change stylesheet or FreeCAD theme. Then until this widget is recreated it will not be properly recalculated. 

You also do not pass option to it so it may not properly react to QSS changes especially pseudo classes (states) like :hover. I think that it would be better to repeat that call - it should be pretty cheap as functions like this are intended to be called from paint method.